### PR TITLE
Allow multi-arch manifest-list from `make container`

### PR DESCRIPTION
**Pull request by kaovilai**
by changing output type to image.

Then you can execute command like so to create a multi-arch image
```
BUILDX_PLATFORMS=linux/amd64,linux/arm64 make container
```

When output type is docker, only single arch is supported and following error is observed.
```
ERROR: docker exporter does not currently support exporting manifest lists
```

Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by mateusoliveira43 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -100,7 +100,7 @@ RESTIC_VERSION ?= 0.15.0
 
 CLI_PLATFORMS ?= linux-amd64 linux-arm linux-arm64 darwin-amd64 darwin-arm64 windows-amd64 linux-ppc64le
 BUILDX_PLATFORMS ?= $(subst -,/,$(ARCH))
-BUILDX_OUTPUT_TYPE ?= docker
+BUILDX_OUTPUT_TYPE ?= image
```
[the remark this answers is not in the document]
Comment (kaovilai) [reply Makefile:103]: Nope. OCI requires destination file, not a container image.

```
ERROR: dest file is required for oci exporter. refusing to write to console
make: *** [container] Error 1
```

Comment (blackpiglet) [inline Makefile:103]: https://docs.docker.com/build/exporters/image-registry/
According to the document, we should add parameters to let the image be pushed to destinations automatically.
For example `BUILDX_OUTPUT_TYPE ?= image,push=true`.

The original `BUILDX_OUTPUT_TYPE ?= docker` will push the images after build.

Comment (kaovilai) [reply Makefile:103]: > The original BUILDX_OUTPUT_TYPE ?= docker will push the images after build.

I didn't find that [to be the case in the docs](https://docs.docker.com/build/exporters/#:~:text=Exporters%20save%20your,creates%20a%20cache.).. The only thing that would push automatically is 

```
--output=type=registry
```

> Exporters save your build results to a specified output type. You specify the exporter to use with the [--output CLI option](https://docs.docker.com/reference/cli/docker/buildx/build/#output). Buildx supports the following exporters:
> 
> image: exports the build result to a container image.
> registry: exports the build result into a container image, and pushes it to the specified registry.
> local: exports the build root filesystem into a local directory.
> tar: packs the build root filesystem into a local tarball.
> oci: exports the build result to the local filesystem in the [OCI image layout](https://github.com/opencontainers/image-spec/blob/v1.0.1/image-layout.md) format.
> docker: exports the build result to the local filesystem in the [Docker Image Specification v1.2.0](https://github.com/moby/moby/blob/v25.0.0/image/spec/v1.2.md) format.
> cacheonly: doesn't export a build output, but runs the build and creates a cache.

Comment (kaovilai) [reply Makefile:103]: > should add parameters to let the image be pushed to destinations automatically.
For example BUILDX_OUTPUT_TYPE ?= image,push=true

will do.

Comment (kaovilai) [reply Makefile]: done via https://github.com/vmware-tanzu/velero/pull/8252/commits/d97885b6c3c3d93a3db4cec09a035f00cdd49129

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**